File: comparisons/categorical_anova.py
import pandas as pd
import statsmodels.formula.api as smf

print("\n=== Python (statsmodels) ===")
data = pd.read_csv("tests/data/pastes.csv")

# statsmodels automatically dummifies categorical variables in strings
model = smf.mixedlm("strength ~ C(cask)", data, groups=data["batch"])
result = model.fit(reml=True)
print(result.summary())

# Multi-Dof Wald ANOVA isn't generically available as a single function call in statsmodels for mixedlm without custom contrast matrices,
# but we can test the joint hypothesis of the 'cask' parameters being zero.
print("\n=== Wald Joint Test for cask ===")
wald_test = result.wald_test("C(cask)[T.b] = 0, C(cask)[T.c] = 0")
print(wald_test)

File: comparisons/categorical_anova.jl
using DataFrames
using CSV
using MixedModels
using Printf

println("\n=== Julia (MixedModels.jl) ===")
const data = DataFrame(CSV.File("tests/data/pastes.csv"))

# MixedModels automatically detects string columns as categorical
model = fit(MixedModel, @formula(strength ~ cask + (1 | batch)), data, REML=true)
println(model)

# MixedModels automatically provides Wald tests for all components? Not natively as a unified DataFrame block without external packages.
